cli/js/40_jupyter.js

```diff
@@ -336,8 +356,90 @@ async function formatInner(obj, raw) {
 
 internals.jupyter = { formatInner };
 
+/**
+class CommMessage(TypedDict):
+    header: dict
+    # typically UUID, must be unique per message
+    msg_id: str
+    msg_type: str
+    parent_header: dict
+    metadata: dict
+    content: <custom payload>
+    buffers: list[memoryview]
+
+((async) => {
+  const data = await Deno.jupyter.comms.recv("1234-5678");
+})();
+((async) => {
+  const data = await Deno.jupyter.comms.recv("1234-5678");
+})();
+
+const comm = await Deno.jupyter.comms.open("1234-5678");
+const data = await comm.recv();
+
+const data = await Deno.jupyter.comms.recv("1234-5678");
+
+c = Comm("1234-5678")
+
+c.on("update", data => {
+    console.log(data);
+    Deno.jupyter.broadcast(...);
+});
+
+
+{
+    msg_type: "comm_msg",
+    content: {
+        comm_id: "1234-5678",
+        data: {
+
+        }
+    }
+}
+*/
+
 function enableJupyter() {
-  const { op_jupyter_broadcast } = core.ops;
+  const { op_jupyter_broadcast, op_jupyter_comm_recv, op_jupyter_comm_open } =
+    core.ops;
+
+  function comm(commId, targetName, msgCallback) {
+    op_jupyter_comm_open(commId, targetName);
+
+    let closed = false;
+
+    // TODO(bartlomieju): return something, so we can close this comm.
+    (async () => {
+      while (true) {
+        const [data, buffers] = await op_jupyter_comm_recv(commId);
+
+        if (!data) {
+          closed = true;
+          break;
+        }
+
+        msgCallback?.({
+          ...data,
+          buffers,
+        });
```

Comment:
Follow up on comment below, to mirror the send API 

```suggestion
        msgCallback?.(data, buffers);
```